### ISSUE TITLE
[Bug]:  OverflowMenu component should have a role

### ISSUE THREAD

**Issue by Sebastian-Ribeiro** [carbon-design-system/carbon#11746]
### Package

carbon-components-react

### Browser

Chrome

### Package version

v7.56.0

### Description

The OverflowMenu component should have a role assigned as an accessibility issue.

### Reproduction/example

https://codesandbox.io/s/optimistic-bhabha-zs3tq3

### Steps to reproduce

- Create a component using the OverflowMenu
- Check in the inspect mode that the object does not have a role

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
